addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/AbstractKNXChannelType.java

```diff
@@ -0,0 +1,194 @@
+            ChannelConfiguration conf = parse((String) channelConfiguration.get(key));
+            if (conf != null) {
+                ret.addAll(
+                        conf.getReadGAs().stream().map(this::toGroupAddress).filter(Objects::nonNull).collect(toSet()));
+            }
+        }
+        return ret;
+    }
+
+    public final Set<GroupAddress> getWriteAddresses(Configuration channelConfiguration) {
+        Set<GroupAddress> ret = new HashSet<>();
+        for (String key : getAllGAKeys()) {
+            ChannelConfiguration conf = parse((String) channelConfiguration.get(key));
+            if (conf != null && conf.getMainGA() != null) {
+                GroupAddress ga = toGroupAddress(conf.getMainGA());
+                if (ga != null) {
+                    ret.add(ga);
+                }
+            }
+        }
+        return ret;
+    }
+
+    private @Nullable GroupAddress toGroupAddress(@Nullable GroupAddressConfiguration ga) {
+        if (ga != null) {
+            try {
+                return new GroupAddress(ga.getGA());
+            } catch (KNXFormatException e) {
+                logger.warn("Could not parse group address '{}'", ga.getGA());
+            }
+        }
+        return null;
+    }
+
+    protected final Set<GroupAddress> getAddresses(@Nullable Configuration configuration, Iterable<String> addresses)
+            throws KNXFormatException {
+        Set<GroupAddress> ret = new HashSet<>();
+        for (String address : addresses) {
+            if (configuration != null && configuration.get(address) != null) {
+                ret.add(new GroupAddress((String) configuration.get(address)));
+            }
+        }
+        return ret;
+    }
+
+    protected final boolean isEquals(@Nullable Configuration configuration, String address, GroupAddress groupAddress)
+            throws KNXFormatException {
+        if (configuration != null && configuration.get(address) != null) {
+            return Objects.equals(new GroupAddress((String) configuration.get(address)), groupAddress);
+        }
+        return false;
+    }
+
+    public final @Nullable OutboundSpec getCommandSpec(Configuration configuration, KNXTypeMapper typeHelper,
+            @Nullable Type command) throws KNXFormatException {
+        for (String key : getAllGAKeys()) {
+            ChannelConfiguration config = parse((String) configuration.get(key));
+            if (config != null) {
+                String dpt = config.getDPT();
+                if (dpt == null) {
+                    dpt = getDefaultDPT(key);
+                }
+
+                Class<? extends Type> expectedTypeClass = typeHelper.toTypeClass(dpt);
+                if (expectedTypeClass != null) {
+                    logger.trace("The expected Type class for dpt '{}' is '{}'", dpt,
+                            expectedTypeClass.getSimpleName());
+                    Type convertedType = convertType(command, configuration);
```

Comment:
How is this related to the "ETS" binding? Aren't you changing the basic binding's behavior here?

Don't get me wrong, potentially this makes sense. I'm just not quite sure yet about the side-effects this might have. In any case, I'm just trying to understand the reason for this change.

Reply:
For the basic binding this function returns the Type is was fed without any modification. In the advanced binding I overload it to do something meaningful. See for example TypeRollerShutter.java on how I do use this functionality.